### ISSUE TITLE
JDK 9 (build 9-ea+132) class loading log not recognized

### ISSUE THREAD

**Issue by msteindorfer** [AdoptOpenJDK/jitwatch#234]
I saw that there were two previous issues about that topic (issue #168, #122), but it seems that the latest log file formats of JDK 9 are currently not supported.

When running JITwatch with the required standard flag (cf. README), I still get the prompt that I should run the program again with `-XX:+TraceClassLoading` although I did.

Looking at the standard output, I found the following message: 

```
# [0.005s][warning][arguments] -XX:+TraceClassLoading is deprecated. Will use -Xlog:class+load=info instead.
```

And further research revealed that indeed the logging of class loading was unified, see the following links:
- http://openjdk.java.net/jeps/158
- http://hg.openjdk.java.net/jdk9/jdk9/hotspot/rev/3637ec3e50c2

It seems that by default `Xlog:class+load=info` logs class loading output just to stdout. There are options to log it to a file as well. However I didn't figure out how I can obtain a singular file that is consumable with JITwatch.

**Comment by chriswhocodes** [comment]
Hi Michael,

I would have to say that JITWatch is not currently compatible with JDK9.

As I understand it, LogCompilation output is mostly the same, but I'm currently prioritising new JITWatch functionality over JDK9 support until the Unified Logging and project Jigsaw changes have stabilised.

I will leave this issue open as a reminder.

Kind regards,

Chris

**Comment by msteindorfer** [comment]
Thanks for the answer, Chris. It's completely understandable that you want to wait until JDK 9 features stabilize.

For the meantime, can you recommend another set of tools for digging deeper into JDK 9 performance? I'm not aware of compelling alternatives to the excellent JITwatch.

**Comment by chriswhocodes** [comment]
Hi Michael,

As a stop-gap measure it's not too hard to follow the LogCompilation output by hand:

Each compilation event (apart from compile_kind='c2n' native code stubs which have no task tag) has the following 3 tag sequence (usually spaced far apart in the log):
```xml
<task_queued compile_id='ID' method='SIGNATURE' ... />
<nmethod compile_id='ID' method='SIGNATURE' ... />
<task compile_id='ID' method='SIGNATURE' ... />
  <!-- interesting stuff in here about inlining, branches, escape analysis etc -->
</task>
```
So you can follow the events for a method with a bit of grepping.

I do have a plan to make JITWatch work without classloading statements and build a class model using only what's in the LogCompilation. That will help when classes aren't available anyway (JITWatch as a webservice etc).

Will see what progress I can make on that :)

**Comment by chriswhocodes** [comment]
Initial JDK9 support added with https://github.com/AdoptOpenJDK/jitwatch/commit/3b2d1ae4dadaf9088d9be7c32502d0b0fa199876.

With JDK9 it no longer tries to "learn" the classpath from the classloader debug in the Hotspot log so you'll need to mount the jars and folders in the main config.
